tests/unit-tests/api/utilities/A_RenderFunctionTest.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Beans\BeansFramework\Tests\API\Utilities;
+
+use Beans\BeansFramework\Tests\BeansTestCase;
+
+class ARenderFunctionTest extends BeansTestCase {
+
+	protected function setUp() {
+		parent::setUp();
+		require_once BEANS_TESTS_LIB_DIR . '/api/utilities/functions.php';
+	}
+
+	public function testRenderFunctionBailsForNonCallable() {
+		$this->assertNull( beans_render_function( 'this-callback-doesnot-exist' ) );
+	}
+
+	public function testRenderFunction() {
+		$this->assertSame(
+			'Hey, no args passed to me.',
+			beans_render_function( array( $this, 'renderFunctionCallback' ) )
+		);
+		$this->assertSame(
+			'foo',
+			beans_render_function( array( $this, 'renderFunctionCallback' ), 'foo' )
+		);
+		$this->assertSame(
+			'foo, bar, baz',
+			beans_render_function( array( $this, 'renderFunctionCallback' ), 'foo', 'bar', 'baz' )
+		);
+		$this->assertSame(
+			'foo, bar, baz',
+			beans_render_function( array( $this, 'renderFunctionCallback' ), array( 'foo', 'bar' ), 'baz' )
+		);
+
+		$this->assertSame(
+			'Hey, no args passed to me.',
+			beans_render_function( __NAMESPACE__ . '\renderFunctionCallback' )
+		);
+		$this->assertSame(
+			'foo',
+			beans_render_function( __NAMESPACE__ . '\renderFunctionCallback', 'foo' )
+		);
+		$this->assertSame(
+			'foo, bar, baz',
+			beans_render_function( __NAMESPACE__ . '\renderFunctionCallback', 'foo', 'bar', 'baz' )
+		);
+		$this->assertSame(
+			'foo, bar, baz',
+			beans_render_function( __NAMESPACE__ . '\renderFunctionCallback', array( 'foo', 'bar' ), 'baz' )
+		);
+	}
+
+	/**************************************
+	 * Helpers
+	 *************************************/
+
+	public function renderFunctionCallback() {
```

Comment:
Why not mock these callback functions, instead of actually defining them?

Reply:
@GaryJones I wanted to ensure that the callback is truly called and that the arguments are being passed.  While we can mock it out, you'll see that I'm doing some processing work to further test the expected behavior.